src/stores/activitiesStores/ActivityHistoryStore.js

```diff
@@ -0,0 +1,104 @@
+import { observable, action, reaction, computed } from 'mobx';
+import { TransactionType, SortBy, Routes } from 'constants';
+import _ from 'lodash';
+
+import { getDetailPagePath } from '../../helpers/utility';
+import { queryAllTransactions, queryAllOracles } from '../../network/graphQuery';
+import Transaction from '../models/Transaction';
+import Oracle from '../models/Oracle';
+
+const INIT_VALUES = {
+  transactions: [],
+  order: SortBy.DESCENDING.toLowerCase(),
+  orderBy: 'createdTime',
+  perPage: 10,
+  page: 0,
+  limit: 50,
+};
+
+export default class {
+  @observable transactions = INIT_VALUES.transactions
+  @observable order = INIT_VALUES.order
+  @observable orderBy = INIT_VALUES.orderBy
+  @observable perPage = INIT_VALUES.perPage
+  @observable page = INIT_VALUES.page
+  @observable limit = INIT_VALUES.limit
+
+  constructor(app) {
+    this.app = app;
+    reaction( // Sort while order changes
```

Comment:
I still think we can combine this with the reaction below that's gettingMoreData

Reply:
After trying I think it's not that easy combine them. I guess better to just leave it there.